### DISCUSSION TITLE
Mbuffer error

### DISCUSSION THREAD

**Discussion by gpetrom** [psy0rz/zfs_autobackup#240] [General]
Hi 

I am testing zfs-autobackup and i was playing with mbuffer. When i run the following command following the instructions from the manual
`
zfs-autobackup -v --clear-mountpoint --ssh-target debian11_test offsite1 storage --no-holds --strip-path=1 --keep-source=30s3m --keep-target=30s3min --min-change=0 --snapshot-format={}-%Y-%m-%d_%H:%M:%S --buffer 128k
`
I am getting that the following  error
`
! [Source] STDERR > mbuffer: fatal: Number of blocks must be at least 5.
! [Target] STDERR > mbuffer: fatal: Number of blocks must be at least 5.
! [Target] STDERR > cannot receive: failed to read from stream
! [Source] Command "zfs send -L -e -v -P -p -i @offsite1-2024-01-14_11:48:03 storage@offsite1-2024-01-14_11:55:44 | mbuffer -q -s128k -m128k" returned exit code 1 (valid codes: [0])
! [Target] Command "ssh debian11_test 'mbuffer -q -s128k -m128k | zfs recv -u -o canmount=noauto -v -s storage'" returned exit code 1 (valid codes: [0])
! [Source] storage: FAILED: Last command returned error
`
Some more questions, is there a way to change mbuffer  -m option? 
I have red that in this option i could put 1G and i will see better transfers. Is this true?
What it is your opinion about mbuffer values?

Thanks in advanced and sorry if my english are bad

**Comment by gpetrom** [comment]
Hi

My bad in the manual you say 128M and not 128K. Sorry about that.
So maybe someone can help me with the other questions. 
Is there a way to change mbuffer -m option?
I have red that in this option i could put 1G and i will see better transfers. Is this true?
What it is your opinion about mbuffer values?

**Comment by psy0rz** [comment]
the --buffer option IS what changes the -m option for the mbuffer command.
